subsys/nrf_security/src/drivers/cracen/cracenpsa/src/common.c

```diff
@@ -125,27 +125,27 @@ static psa_status_t get_sx_brainpool_curve(size_t curve_bits, const struct sx_pk
 
 	switch (curve_bits) {
 	case 192:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP192r1));
 		break;
 	case 224:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP224r1));
 		break;
 	case 256:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_256,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP256r1));
 		break;
 	case 320:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP320r1));
 		break;
 	case 384:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_384,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP384r1));
 		break;
 	case 512:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_512,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP512r1));
```

Comment:
Let's not repeat ourselves so much.

if(IS_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY)) {
switch(curve_bits) bla bla
} else {
  return not supported;
}

Reply:
I would leave the code here referring to NCSDK-26237 as then we have to add it again

Reply:
true